aiohttp/client_reqrep.py

```diff
@@ -1020,6 +1025,12 @@ async def start(self, connection: "Connection") -> "ClientResponse":
                     self._continue = None
 
         # payload eof handler
+        if self.__writer is not None:
+            # Make sure to tell the writer that the reader is done
+            # so it doesn't close the connection if it gets cancelled
+            # because there is still payload being read.
+            self.__writer._reader_done = True
```

Comment:
The problem is that if we don't really know how much is left to be written, and once we write to the buffer we don't know if we can reuse it or now because we might be writing into the next request.

This is a bigger synchronization problem in general